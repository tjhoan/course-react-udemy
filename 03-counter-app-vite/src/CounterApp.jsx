import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleCounterChange = (delta) => {
    setCounter((prevCounter) => prevCounter + delta);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={() => handleCounterChange(1)}> +1 </button>
      <button onClick={() => handleCounterChange(-1)}> -1 </button>
      <button aria-label="btn-reset" onClick={() => setCounter(value)}>
        {' '}
        Reset{' '}
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
};
