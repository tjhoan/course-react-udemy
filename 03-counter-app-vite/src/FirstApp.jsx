import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ title, subTitle = 123, arrayNumbers = [1, 2, 3] }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subTitle}</h2>
      <p>{arrayNumbers.join(' ')}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  arrayNumbers: PropTypes.array
};

export { FirstApp };
