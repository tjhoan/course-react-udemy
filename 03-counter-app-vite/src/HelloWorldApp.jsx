export const HelloWorldApp = () => {
  const getSaludo = (name) => {
    return `Hola ${name}`;
  }

  return (
    <>
      <h1>Hola Mundo</h1>
      <p>Soy un subtitulo</p>
      <p>{getSaludo('Alejandro')}</p>
    </>
  );
};