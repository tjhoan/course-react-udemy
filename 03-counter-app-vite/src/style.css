html,
body {
  background-color: #21232a;
  color: white;
  font-family: Helvetica, sans-serif, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  background-color: #61dafb;
  border-radius: 10px;
  color: #282c34;
  font-size: 1.3rem;
  padding: 15px;
  margin-right: 7px;
}
