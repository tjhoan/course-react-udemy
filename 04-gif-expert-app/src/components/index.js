export * from './AddCategory2';
export * from './GifGrid';
export * from './GifItem';
