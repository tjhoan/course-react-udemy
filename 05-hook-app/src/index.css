body {
  padding: 25px;
  background-color: #414141;
  color: rgb(201, 201, 201);
}

button {
  margin-right: 10px;
}