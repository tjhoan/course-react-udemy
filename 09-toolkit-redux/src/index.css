body {
  background-color: #242424;
  color: white;
  padding: 50px 25px;
  margin: 0;
}

/* Loading */

.loading-container {
  height: 200px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.spinner {
  width: 70px;
  text-align: center;
}

.spinner > div {
  width: 18px;
  height: 18px;
  background-color: #007bff;
  border-radius: 100%;
  display: inline-block;
  animation: bouncedelay 1.4s infinite ease-in-out both;
}

.spinner .bounce1 {
  animation-delay: -0.32s;
}

.spinner .bounce2 {
  animation-delay: -0.16s;
}

@keyframes bouncedelay {
  0%,
  80%,
  100% {
    transform: scale(0);
  }
  40% {
    transform: scale(1);
  }
}

::-webkit-scrollbar {
  width: 11px;
  height: 8px;
}

::-webkit-scrollbar-thumb {
  background: #555;
  border-radius: 10px;
}

::-webkit-scrollbar-thumb:hover {
  background: #888;
}

::-webkit-scrollbar-track {
  background: #222;
  border-radius: 4px;
}