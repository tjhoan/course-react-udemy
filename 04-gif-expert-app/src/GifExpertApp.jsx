import { useState } from 'react';
import { AddCategory2, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = newCategory => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory2 onNewCategory={value => onAddCategory(value)} />
      <ol>
        {categories.map(category => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
