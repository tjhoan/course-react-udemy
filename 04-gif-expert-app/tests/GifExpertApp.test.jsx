import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
  const value = 'Saitama';

  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de cambiar el texto en el input', () => {
    render(<GifExpertApp />);

    const input = screen.getByRole('textbox');
    fireEvent.input(input, { target: { value } });
    expect(input.value).toBe('Saitama'); // Verifica que el valor del input sea 'Saitama'
  });

  test('Debe mostrar la nueva categoría ingresada en el input', () => {
    render(<GifExpertApp />);
    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input(input, { target: { value } }); // Simula el ingreso de un valor en el input
    fireEvent.submit(form); // Simula el envío del formulario

    expect(screen.getByText('Saitama')); // Verifica que el texto 'Saitama' se encuentre en el documento
  });

  test('No debe guardar la categoría si ya existe', () => {
    render(<GifExpertApp />);
    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input(input, { target: { value } });
    fireEvent.submit(form);

    expect(screen.getByText('Saitama')); // Verifica que el texto 'Saitama' se encuentre en el documento

    fireEvent.input(input, { target: { value } });
    fireEvent.submit(form);

    const categories = screen.getAllByText('Saitama'); // Verifica que haya un solo elemento con el texto 'Saitama'
    expect(categories.length).toBe(1); // Verifica que haya un solo elemento con el texto 'Saitama'
  });
});
