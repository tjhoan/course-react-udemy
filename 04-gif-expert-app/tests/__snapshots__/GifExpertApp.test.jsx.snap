// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <ol>
    <h3>
      Cargando...
    </h3>
    <h3>
      One Punch
    </h3>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
